### DISCUSSION TITLE
FirefoxでアニメーションGIFが再生されない

### DISCUSSION THREAD

**Discussion by sonota88** [increments/qiita-discussions#196] [要望・アイデア・不具合]
# 現象
Firefox で記事ページを表示した際、アニメーションGIFが再生されません。

# 例
https://qiita.com/sonota88/items/a5d6d3539e0fb8040f74
（追記 2022-08-27: こちらは編集して保存しなおしたら再生されるようになりました）

https://qiita.com/yhara/items/f42571f5302fa02a3c12

# 環境
Ubuntu 18.04 + Firefox 98.0.2
Windows 10 Home 20H2 + Firefox 98.0.2

# 備考

- 再生されない場合でも、画像をクリックすると、画像のみ別タブで表示されアニメーションが再生される
- 記事編集ページのプレビューでは再生される
- 記事編集して保存し、記事ページに遷移した場合は再生される
  - 別のタブで同じ記事ページを開き直した場合は再生されない

----

再生される画像のURL例
（編集ページのプレビュー、編集後の記事ページのものはこちら）
https://camo.qiitausercontent.com/a1f74982b88310cd9bf52e571fe67f1b33e02cd7/68747470733a2f2f71696974612d696d6167652d73746f72652e73332e61702d6e6f727468656173742d312e616d617a6f6e6177732e636f6d2f302f3233343035352f35613737393832352d613235322d616164312d326161622d3764353166643730383336332e676966

再生されない画像のURL例
（記事ページを普通に開いた場合はこちら）
https://qiita-user-contents.imgix.net/https%3A%2F%2Fqiita-image-store.s3.ap-northeast-1.amazonaws.com%2F0%2F234055%2F5a779825-a252-aad1-2aab-7d51fd708363.gif?ixlib=rb-4.0.0&auto=format&gif-q=60&q=75&s=3bf852c84ba84c1efb2bb3621a082f4d

----

- imgix が返す画像データのフォーマットは GIF ではなく AVIF になっているようです
- AVIF のアニメーション機能は Firefox ではまだサポートされていないようです
  - 1686338 - AVIF (AV1 Image File Format): image sequence and animation support
    https://bugzilla.mozilla.org/show_bug.cgi?id=1686338

----

参考:
Qiitaの記事画像が最適化され、35%転送量が削減されました - Qiita
https://qiita.com/hareku/items/87a8308debe4eddb2940

**Comment by kskwtnk** [comment]
@sonota88 
ご意見ありがとうございます！
環境や調査結果まで詳細に記載いただき、非常に助かります！

確認したところ、解消方法がすぐには見つからず……調査いたしますので今しばらくお待ちください 🙇

**Reply by sonota88** [reply]
お返事ありがとうございます！

記事を書く側で取れる対策はアニメーションである旨の注釈を付けるといったものに限られますので、ご対応いただけますと大変助かります :bow:

**Comment by tsuzuki-takaaki** [comment]
@sonota88
報告が遅くなってしまい、申し訳ございません。

こちらの不具合ですが、ご記載いただいている通り、Imgix が返す AVIF 形式に過去の Firefox が対応していなかったことが原因でした。
[Firefox 113.0](https://www.firefox.com/en-US/firefox/113.0/releasenotes/) で AVIF 形式に対応されているため、それ以降のバージョンの Firefox を使っていただけると解消すると思われます。

合わせて、下記の discussion もご確認いただけますと幸いです。
https://github.com/increments/qiita-discussions/discussions/422

よろしくお願いいたします🙇

**Reply by sonota88** [reply]
ご連絡ありがとうございます。

> https://qiita.com/sonota88/items/a5d6d3539e0fb8040f74
> （追記 2022-08-27: こちらは編集して保存しなおしたら再生されるようになりました）

この後、今度は画像自体表示されなくなってしまいましたが（Chrome でも表示されない）、アップロードしなおすなど再度試してみようと思います。
